cloud/services/groups/groups.go

```diff
@@ -55,10 +55,16 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	ctx, span := tele.Tracer().Start(ctx, "groups.Service.Reconcile")
 	defer span.End()
 
-	if _, err := s.client.Get(ctx, s.Scope.ResourceGroup()); err == nil {
+	_, err := s.client.Get(ctx, s.Scope.ResourceGroup())
+	if err == nil {
```

Comment:
isn't this equivalent to what we were doing before with `if _, err := s.client.Get(ctx, s.Scope.ResourceGroup()); err == nil {` ?

Reply:
Yes, but err will be local to `if _, err := s.client.Get(ctx, s.Scope.ResourceGroup()); err == nil {` block.
We will need the err in `if !azure.ResourceNotFound(err) {
 		return err
 	}
`
I will modify the PR with an else if block, to minimise the number of lines changed.